File: DemoMethods/Basic/DynamicAggregation.cs
using System;
using System.Web.Http;
using DemoMethods.Entities;
using DemoMethods.Helpers;
using DemoMethods.Indexes;
using Raven.Client;
using Raven.Client.Linq;

namespace DemoMethods.Basic
{
    public partial class BasicController : DemoApiController
    {
        [HttpGet]
        [Demo("Dynamic Aggregation", DemoOutputType.Flatten, demoOrder: 90)]
        public object DynamicAggregation(string fromVal = "0", string toVal = "999")
        {
            var from = int.Parse(fromVal);
            var to = int.Parse(toVal);

            using (var session = DocumentStoreHolder.Store.OpenSession())
            {
                var result = session.Query<Product, Products>()
                    .Where(x => x.UnitsInStock >= from && x.UnitsInStock <= to)
                    .AggregateBy(x => x.Category)
                    .SumOn(x => x.UnitsInStock)
                    .ToList();

                return DemoUtilities.FormatRangeResults(result.Results);
            }
        }
    }
}

File: DemoMethods/src/main/java/net/ravendb/demo/basic/DynamicAggregation.java
package net.ravendb.demo.basic;

import net.ravendb.abstractions.data.FacetResults;
import net.ravendb.client.IDocumentSession;
import net.ravendb.demo.DemoUtilities;
import net.ravendb.demo.DocumentStoreHolder;
import net.ravendb.demo.entities.Product;
import net.ravendb.demo.entities.QProduct;
import net.ravendb.demo.indexes.FacetRangeCreation;
import net.ravendb.demo.indexes.Products;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;

import java.util.List;

@Controller
public class DynamicAggregation {

    @RequestMapping("/Basic/DynamicAggregation")
    public List<FacetRangeCreation.FacetsRangesResults> dynamicAggregation(
            @RequestParam(value = "fromVal", defaultValue = "0") int fromVal,
            @RequestParam(value = "toVal", defaultValue = "999") int toVal) {

        try (IDocumentSession session = DocumentStoreHolder.getStore().openSession()) {
            QProduct p = QProduct.product;
            FacetResults result = session.query(Product.class, Products.class)
                    .where(p.unitsInStock.goe(fromVal).and(p.unitsInStock.loe(toVal)))
                    .aggregateBy(p.category)
                    .sumOn(p.unitsInStock)
                    .toList();

            return DemoUtilities.formatRangeResults(result.getResults());
        }
    }
}
